### ISSUE TITLE
Installing on windows fails. --prefer-binary option for the venv pip is not recognized. Gist with updated installer.py included

### ISSUE THREAD

**Issue by JVanloofsvelt** [lektor/lektor#808]
On Windows, the virtual environment created by installer.py comes with an old pip version. It complains about the option '--prefer-binary' since this option was not supported by the old pip version.

It might be related to the fact that I have an older Python 3.5 installation that came with an older pip version.

**Comment by JVanloofsvelt** [comment]
I did some research, apparently the installer has three strategies for creating a virtualenv:
1. using the builtin venv module
2. using virtualenv from site-packages if venv was not an option
3. downloading virtualenv if virtualenv was not found

Remarks:
- the built-in 'venv' module uses the builtin 'ensurepip' module, which comes bundled with a pip version. This version cannot be updated, it is supposed to be frozen with the specific Python distribution/version.
- the '--prefer-binary' option for pip was added in pip version 18.0

How should the installer.py be modified then?
IMO: use_venv() should fail (or rather be skipped) if the pip version that comes with ensurepip() is older than version 18.0.

The version of the pip bundled with ensurepip can be queried like this:
```
import ensurepip
ensurepip_version = ensurepip.version()
````
So if the returned version is less than 18.0, then 'use_venv()' can return None at this point, so that the next strategy for creating a virtualenv is used (which is using virtualenv from site-packages, and that one CAN be upgraded independently of the Python distribution). 

I cannot find the installer.py in this repository, so I can't create a pull request.
